dir-spec.txt

```diff
@@ -1588,6 +1588,14 @@
    and sending it in an HTTP POST request to each other authority at the URL
      http://<hostname>/tor/post/vote
 
+   If an authority has used a Bandwidth List file to generate this vote it
+   SHOULD make it available at
+
+     http://<hostname>/tor/status-vote/next/bandwidth.z
+
+   If an authority makes this file available, its headers MUST correspond to
+   the ones in the "bandwidth-file-headers" item in the vote document.
```

Comment:
Don't say "it's headers MUST correspond to the ones in the…vote document", because that's confusing.
It could mean that tor reads the headers from the vote, which is not what you want.
Instead, say "it MUST be the file used to create the vote".

The bandwidth generator can overwrite the bandwidth file at any time.
How are you going to store the old bandwidth file so the authority can serve it with the old vote?

You could say:
"If an authority makes this file available, it MAY be the bandwidth file used to create the vote, or the bandwidth file currently on disk."

Which vote document?
"the vote document available at http://<hostname>/tor/status-vote/next/authority.z"?

Please also reference the bandwidth file spec, so people know the format of the file.

Reply:
Or you can say:
"If an authority makes this file available, it MUST be the bandwidth file used to create the vote."

Then tor MUST make a copy of the file before closing it.
(Tor and sbws will need to be careful of race conditions when writing the file.)

Reply:
Please also say the version of Tor that introduced this feature. Looks like it will be 0.3.5.

Reply:
I opened a sbws ticket to avoid the race condition:
https://trac.torproject.org/projects/tor/ticket/26692

Reply:
> The bandwidth generator can overwrite the bandwidth file at any time.

Torflow archives the file renaming it with a timestamp and putting it in a dir, but scp inside cron.sh sends the file without timestamp. 
This is an easy change so that it sends the file with the timestamp.
In the DirAuth there should be a cron or ssh command that create a link to the latest.
If Torflow runs in the same system as the DirAuth then it could just read the link locally.

sbws already archives files with timestamp, then creates a link to the latest.

> How are you going to store the old bandwidth file so the authority can serve it with the old vote?

i don't understand this question

> "If an authority makes this file available, it MAY be the bandwidth file used to create the vote, or the bandwidth file currently on disk."

I like more the idea of trying to ensure that the file used to create the vote is the one being served.

> "If an authority makes this file available, it MUST be the bandwidth file used to create the vote."

Hmm, maybe it's not needed at all to say this sentence, it's basically the same as the 1st one.

> Then tor MUST make a copy of the file before closing it.

This would be a solution, while is not done in the same path where the generator puts the file.

> (Tor and sbws will need to be careful of race conditions when writing the file.)

Hmm, Tor doesn't need to write to that file, only read, right?.